controllers/dataplane/openstackdataplanenodeset_controller.go

```diff
@@ -369,13 +369,13 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 			condition.InitReason, condition.InitReason)
 
 		// Set Images
-		if containerImages.OsContainerImage != nil {
+		if containerImages.OsContainerImage != nil && instance.Spec.BaremetalSetTemplate.OSContainerImageURL == "" {
```

Comment:
i'm good with the change here to allow overrding the image. related though, does the spec change here actually get saved? i thought updating the spec in the reconcile was a bit of an anti-pattern.

Reply:
Good point.. I think it does get saved, but let's change it to update when creating baremetalset spec.